### PR TITLE
[Vengeance] implement Darkglare Boon

### PR DESCRIPTION
**Pull request by ToppleTheNun**
@EvanMichaels

### CODE REVIEW

File: engine/class_modules/sc_demon_hunter.cpp [hunk 1]
```diff
@@ -2371,6 +2371,25 @@ struct fel_devastation_t : public demon_hunter_spell_t
...
+    {
+      // CDR reduction and Fury refund are separate rolls per Realz
+      timespan_t minimum_cdr_reduction = timespan_t::from_seconds(p()->talent.vengeance.darkglare_boon->effectN(1).base_value());
+      timespan_t maximum_cdr_reduction = timespan_t::from_seconds(p()->talent.vengeance.darkglare_boon->effectN(2).base_value());
+      timespan_t cdr_reduction = rng().range(minimum_cdr_reduction, maximum_cdr_reduction);
+      double minimum_fury_refund = p()->talent.vengeance.darkglare_boon->effectN(3).base_value();
+      double maximum_fury_refund = p()->talent.vengeance.darkglare_boon->effectN(4).base_value();
+      double fury_refund = rng().range(minimum_fury_refund, maximum_fury_refund);
+
+      p()->cooldown.fel_devastation->adjust(cdr_reduction);
```
Comment (EvanMichaels) [inline engine/class_modules/sc_demon_hunter.cpp]: Isn't the adjustment value coming from spell data the percentage of the total cooldown reduced? Or is the tooltip wrong? Either way, this value should be negative though?

Comment (ToppleTheNun) [reply engine/class_modules/sc_demon_hunter.cpp]: That is correct, not a flat value; I'll fix it.

Comment (ToppleTheNun) [reply engine/class_modules/sc_demon_hunter.cpp]: Just a note that a flat value ends up working out atm due to Fel Dev CD being 1min by default.

File: engine/class_modules/sc_demon_hunter.cpp [hunk 2]
```diff
@@ -2372,6 +2372,28 @@ struct fel_devastation_t : public demon_hunter_spell_t
...
+      timespan_t minimum_cdr_reduction =
+          timespan_t::from_seconds( p()->talent.vengeance.darkglare_boon->effectN( 1 ).percent() * base_cooldown );
+      timespan_t maximum_cdr_reduction =
+          timespan_t::from_seconds( p()->talent.vengeance.darkglare_boon->effectN( 2 ).percent() * base_cooldown );
+      timespan_t cdr_reduction   = rng().range( minimum_cdr_reduction, maximum_cdr_reduction );
+      double minimum_fury_refund = p()->talent.vengeance.darkglare_boon->effectN( 3 ).base_value();
+      double maximum_fury_refund = p()->talent.vengeance.darkglare_boon->effectN( 4 ).base_value();
+      double fury_refund         = rng().range( minimum_fury_refund, maximum_fury_refund );
+
+      p()->cooldown.fel_devastation->adjust( cdr_reduction );
```
Comment (EvanMichaels) [inline engine/class_modules/sc_demon_hunter.cpp]: This value needs to be passed in as -cdr_reduction since right now this would add to the cooldown.

**Pull request merged**